### DISCUSSION TITLE
Please solve the cors error on my website

### DISCUSSION THREAD

**Discussion by cupid20103** [community/community#60408] [Programming Help]
### Body

Hello, everyone!

these are github repos of my test projects and they have errors like the images.
[front-end github repo](https://github.com/cupid20103/website-aggregator-client)
[back-end github repo](https://github.com/cupid20103/website-aggregator-server)

This is an image showing the cors error.

![image](https://github.com/community/community/assets/102618556/91ae86ef-79d3-4179-90ec-fee5159b367c)

I deployed my front-end and back-end projects to Vercel.
these are deployed URL on Vercel
[front-end link](https://website-aggregator-client.vercel.app/)
[back-end link](https://website-aggregator-server.vercel.app/)

this is the code snippet for the backend's cors option.
`app.use(cors(corsOption));`

please solve this cors error.
thanks.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by melvincwng** [comment]
Hi @cupid20103, thanks for posting your query on GitHub Community DIscussions.

I've took a look at your backend repository - the issue seems to be with this particular line of code over here: https://github.com/cupid20103/website-aggregator-server/blob/main/index.js#L16.

In short, corsOption is **undefined** in your backend repo, hence the CORS error occurs as seen in the developer console.

To resolve the error, you can do the following:
```js
// Enable cors in your backend application for all routes
const corsOption = {
  origin: true
};

app.use(cors(corsOption));
```
Hope this helps and answers your question 😄, take care and have a good day ahead!

References:
1. https://expressjs.com/en/resources/middleware/cors.html

**Reply by cupid20103** [reply]
Hello, @melvincwng. Thanks for your reply.
I updated the code as you instructed.
But still getting cors error.
![image](https://github.com/community/community/assets/102618556/07b0a6b2-1a0b-4481-8ccd-6b14d1ccd6ea)
I think there is something wrong with vercel.
Anyway, thank you very much 🥰

**Reply by melvincwng** [reply]
Hey @cupid20103, deploying Express.js apps on Vercel requires additional configuration changes in order for the API endpoints to work correctly. Or else, they won't work.

For reference: 
![image](https://github.com/community/community/assets/77479885/c50cc694-5f19-4071-bd67-0329f1bf6f1d)

From your backend repository, seems like the necessary folder structure (i.e. `/api` folder) is missing. That's probably the reason why your API endpoints are still failing.

You might want to read up & check the following references below to see how to properly configure & deploy Express.js apps on Vercel:
1. [Using Express with Vercel](https://vercel.com/guides/using-express-with-vercel)
2. [StackOverFlow - How to get Express.js apps working with Vercel](https://stackoverflow.com/questions/66844611/vercel-express-cant-seem-to-get-any-complex-paths-working)

Alternatively, consider deploying your Express.js apps on alternative platforms like Heroku - which could be easier since there isn't a need to do all these specific configurations or setup, which are unique to Vercel deployment.

Hope this helps and resolves your issue :)

**Comment by cupid20103** [comment]
Ah~ sorry about that.
I forgot to edit the vercel.json file.
We need to make vercel.json file to solve it.
So I just made vercel.json file. 🤗

This is vercel.json file on my project

```
{
  "version": 2,
  "name": "website-aggregator-server",
  "builds": [{ "src": "index.js", "use": "@vercel/node" }],
  "routes": [{ "src": "/(.*)", "dest": "/index.js" }]
}
```

But this time I got the following error.
I don't know why. 😞
Please solve it.

![image](https://github.com/community/community/assets/102618556/478c90c1-c859-47cc-b54f-03f2d30c7492)

**Comment by melvincwng** [comment]
@cupid20103, could you try following the section 'Standalone Express' in this [Vercel Documentation](https://vercel.com/guides/using-express-with-vercel)?

You seem to be still missing an `/api` folder and the vercel.json should follow the format as stated in the documentation.

These configurations are unique to Vercel deployment actually.

**Comment by Beargruug** [comment]
Hey, It could be that you need to pass some more information in your request: e.g 
```
fetch('someURL', 
  { 
    method: "GET", 
    mode: 'cors', 
    headers: { 
      'Content-Type': 'application/json',
    }}).then(response => response.json())
```
try to add the mode for you're case. 
Hope this will help you!
